cmd/crawler/main.go

```diff
@@ -0,0 +1,81 @@
+package main
+
+import (
+    "fmt"
+    "strings"
+    "bufio"
+    "os"
+    "time"
+
+    "github.com/gocolly/colly/v2"
+)
+
+func main() {
+    fmt.Println("Go go yees")
+    c := colly.NewCollector()
+
+    used := make(map[string]bool)
+    urlQueue := make([]string, 0)
+    articles := make([]string, 0)
+
+    c.OnHTML("a[href]", func(e *colly.HTMLElement) {
+        url := e.Attr("href")
+        if !used[url] && strings.Contains(url, "arxiv.org") {
+            if strings.Contains(url, "abs") {
+                articles = append(articles, url)
+            }
+            used[url] = true
+            urlQueue = append(urlQueue, url)
+        }
+	})
+
+    c.OnRequest(func(r *colly.Request) {
+		fmt.Println("Visiting", r.URL)
+	})
+
+    urlQueue = append(urlQueue, "https://arxiv.org/covid19search")
+
+    needArticles := 10
+
+    for len(articles) < needArticles && len(urlQueue) > 0 {
```

Comment:
Looks weird to me.. try to express this intent in a more idiomatic and clear way.

Reply:
Done